pandas/plotting/_matplotlib/core.py

```diff
@@ -1327,7 +1326,14 @@ def __init__(self, data, **kwargs):
                 self.tickoffset = self.bar_width * pos
                 self.lim_offset = 0
 
-        self.ax_pos = self.tick_pos - self.tickoffset
+        if isinstance(self.data.index, ABCMultiIndex):
+            # No real handling for MultiIndex yet
+            warnings.warn(
```

Comment:
do we hit this in tests? why are we not just raising?

Reply:
Good remark. The real behavior that we want to warn against is redrawing a MultiIndex on a existing axis with data. I change the code and the test accordingly, so that drawing a Bar Plot with a MultiIndex does not warn (it is supported, and documented), but that redrawing is not (that is the original bug issue, where reordering the data and replotting on the same axis lead to inconsistent labels.)